qiita_db/metadata_template.py

```diff
@@ -623,23 +619,62 @@ def __setitem__(self, column, value):
         """
         conn_handler = SQLConnectionHandler()
 
-        exists = conn_handler.execute_fetchone("""
+        # try dynamic tables
+        exists_dynamic = conn_handler.execute_fetchone("""
+        SELECT EXISTS (
             SELECT column_name
             FROM information_schema.columns
             WHERE table_name='{0}'
                 AND table_schema='qiita'
-                AND column_name='{1}'""".format(self._dynamic_table, column))
+                AND column_name='{1}')""".format(self._dynamic_table,
+                                                 column))[0]
+        # try required_sample_info
+        exists_required = conn_handler.execute_fetchone("""
+        SELECT EXISTS (
+            SELECT column_name
+            FROM information_schema.columns
+            WHERE table_name='required_sample_info'
+                AND table_schema='qiita'
+                AND column_name='{0}')""".format(column))[0]
 
-        if exists is None:
+        if exists_dynamic:
+            # catching error so we can check if the error is due to different
+            # column type or something else
+            try:
+                conn_handler.execute("""
+                    UPDATE qiita.{0}
+                    SET {1}=%s
+                    WHERE sample_id=%s""".format(self._dynamic_table,
+                                                 column), (value, self._id))
+            except Exception as e:
+                column_type = conn_handler.execute_fetchone("""
+                    SELECT data_type
+                    FROM information_schema.columns
+                    WHERE column_name=%s AND table_schema='qiita'
+                    """, (column,))[0]
+                value_type = type(value).__name__
+
+                if column_type != value_type:
+                    raise ValueError("""The new value being added to column:
```

Comment:
Changing but not really as in html multiple spaces are changed to one single space.